app/src/main/java/com/andrew/tasky/agenda/data/event/EventRepositoryImpl.kt

```diff
@@ -32,116 +33,199 @@ class EventRepositoryImpl @Inject constructor(
+        }
+        val updatedEvent = event.copy(photos = updatedPhotos)
+        db.getEventDao().upsertEvent(updatedEvent.toEventEntity())
+        db.getEventDao().upsertModifiedEvent(
+            ModifiedEventEntity(
+                id = updatedEvent.id,
+                modifiedType = ModifiedType.UPDATE
+            )
+        )
+    }
+
+    private suspend fun updateRemoteEventSuccess(
+        result: Resource<EventDto>,
+        event: Event
+    ) {
+        upsertApiResultToDb(result, event)
+        event.photos.filterIsInstance<LocalPhoto>().filter {
+            it.savedInternally
+        }.forEach {
+            imageStorage.deleteImage(it.key)
+        }
+    }
+
+    private suspend fun createRemoteEventSuccess(
+        result: Resource<EventDto>,
+        newEvent: Event
+    ) {
+        upsertApiResultToDb(result, newEvent)
+    }
+
+    private suspend fun upsertApiResultToDb(
+        result: Resource<EventDto>,
+        event: Event
+    ) {
+        result.data?.let {
+            db.getEventDao().upsertEvent(
+                it.toEventEntity(
+                    isDone = event.isDone,
+                    isGoing = event.isGoing
+                )
+            )
+        }
+    }
+
+    private suspend fun createRemoteEventError(
+        newEvent: Event
+    ) {
+        db.getEventDao().upsertEvent(newEvent.toEventEntity())
+        db.getEventDao().upsertModifiedEvent(
+            ModifiedEventEntity(
+                id = newEvent.id,
+                modifiedType = ModifiedType.CREATE
+            )
+        )
+        newEvent.photos.filterIsInstance<LocalPhoto>().forEach {
+            saveLocalPhotoInternally(it.copy(savedInternally = true))
+        }
+    }
+
+    private suspend fun updateRemoteEvent(event: Event): Resource<EventDto> {
+        val result = getResourceResult {
+            api.updateEvent(
+                eventData = MultipartBody.Part
+                    .createFormData(
+                        name = "update_event_request",
+                        value = Json.encodeToString(event.toUpdateEventRequest())
+                    ),
+                photoData = event.photos.filterIsInstance<LocalPhoto>()
+                    .mapIndexedNotNull { index, eventPhoto ->
+                        MultipartBody.Part
+                            .createFormData(
+                                name = "photo$index",
+                                filename = eventPhoto.key,
+                                body = eventPhoto.byteArray?.toRequestBody()
+                                    ?: return@mapIndexedNotNull null
                             )
-                        )
                     }
-                    compressedEvent.photos.filterIsInstance<LocalPhoto>().filter {
-                        it.savedInternally
-                    }.forEach {
-                        imageStorage.deleteImage(it.key)
+            )
+        }
+        return result
+    }
```

Comment:
If all you do is create a variable and then return it without using it for anything, I'd just return the value right away without creating the variable :)